From 94d65112e421a7198222c0eab2636fc8e2deae7d Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Thu, 28 Aug 2014 14:23:24 -0400
Subject: [cpufreq] Return error if ->get() failed in cpufreq_update_policy()

Message-id: <1409235879-1283-111-git-send-email-prarit@redhat.com>
Patchwork-id: 89427
O-Subject: [RHEL7.1 PATCH BZ 1134639 110/185] cpufreq: Return error if ->get() failed in cpufreq_update_policy()
Bugzilla: 1134369
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Mark Langsdorf <mlangsdo@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1134639

commit bd0fa9bb455d9b58ec2d7a36cd08afed9e3411b4
Author: Viresh Kumar <viresh.kumar@linaro.org>
Date:   Tue Feb 25 14:29:44 2014 +0530

    cpufreq: Return error if ->get() failed in cpufreq_update_policy()

    cpufreq_update_policy() calls cpufreq_driver->get() to get current
    frequency of a CPU and it is not supposed to fail or return zero.
    Return error in case that happens.

    Signed-off-by: Viresh Kumar <viresh.kumar@linaro.org>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Lenny Szubowicz <lszubowi@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/cpufreq/cpufreq.c b/drivers/cpufreq/cpufreq.c
index 140b29b..3b20b79 100644
--- a/drivers/cpufreq/cpufreq.c
+++ b/drivers/cpufreq/cpufreq.c
@@ -2138,6 +2138,11 @@ int cpufreq_update_policy(unsigned int cpu)
   */
  if (cpufreq_driver->get) {
   new_policy.cur = cpufreq_driver->get(cpu);
+  if (WARN_ON(!new_policy.cur)) {
+   ret = -EIO;
+   goto no_policy;
+  }
+
   if (!policy->cur) {
    pr_debug("Driver did not initialize current freq");
    policy->cur = new_policy.cur;
-- 
1.7.1